From 11ee679c536caaa56731998e77257c076e726610 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Fri, 17 Apr 2015 16:11:50 +0530
Subject: [PATCH] XPWR: Changed PMIC interrupt type from edge trigger to level
 trigger.

X-Power PMIC interrupt is active low and irq line will be pulled low
as long as all the irq event are processed and cleared. To handle this
interrupt behavior in driver, the irq flag should be IRQF_TRIGGER_LOW
which triggers driver's ISR continuously  as long as interrupt line
is pulled low to get all the events processed.

Change-Id: I158425c8a17f3cda86bc9005670b3a7c10969f9f
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1434
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/mfd/intel_soc_pmic_dc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mfd/intel_soc_pmic_dc.c b/drivers/mfd/intel_soc_pmic_dc.c
index e2b331a..b1e0b5c 100644
--- a/drivers/mfd/intel_soc_pmic_dc.c
+++ b/drivers/mfd/intel_soc_pmic_dc.c
@@ -668,7 +668,7 @@ static int dollar_cove_init(void)
 
 struct intel_soc_pmic dollar_cove_pmic = {
 	.label		= "dollar cove",
-	.irq_flags	= IRQF_TRIGGER_FALLING | IRQF_ONESHOT
+	.irq_flags	= IRQF_TRIGGER_LOW | IRQF_ONESHOT
 				| IRQF_NO_SUSPEND,
 	.init		= dollar_cove_init,
 	.cell_dev       = dollar_cove_dev,
-- 
1.9.1

